File: centre-api/src/main/java/com/centre/poly/registration/dto/RegistrationDetailsResponse.java
package com.centre.poly.registration.dto;

import com.centre.poly.document.DocumentResponse;
import com.centre.poly.person.entity.Student;
import com.centre.poly.registration.entity.RegistrationStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RegistrationDetailsResponse {
    
    private Long id;
    private RegistrationStatus status;
    private String remarks;
    private LocalDateTime createdDate;
    private Long specialtyId;
    private String specialtyName;
    private Double registrationFees;
    private StudentDetails student;
    private ParentDetails father;
    private ParentDetails mother;
    private Address address;
    private Boolean isAffected;
    private String statusChangeReason;
    private List<DocumentRegistrationResponse> documentRegistrationResponseList;
    
    
}

File: centre-ui/src/app/services/models/registration-details-response.ts
/* tslint:disable */
/* eslint-disable */
import { Address } from '../models/address';
import { DocumentRegistrationResponse } from '../models/document-registration-response';
import { ParentDetails } from '../models/parent-details';
import { StudentDetails } from '../models/student-details';
export interface RegistrationDetailsResponse {
  address?: Address;
  createdDate?: string;
  documentRegistrationResponseList?: Array<DocumentRegistrationResponse>;
  father?: ParentDetails;
  id?: number;
  mother?: ParentDetails;
  registrationFees?: number;
  remarks?: string;
  specialtyId?: number;
  specialtyName?: string;
  status?: 'IN_PROGRESS' | 'COMPLETED' | 'CANCELLED';
  student?: StudentDetails;
}
